nucypher/blockchain/eth/signers/hardware.py

```diff
@@ -277,3 +279,194 @@ def sign_transaction(self,
         if rlp_encoded:
             signed_transaction = HexBytes(rlp.encode(signed_transaction))
         return signed_transaction
+
+
+class HardwareSigner(Signer):
+    """
+    Common base class mixin for hardware wallets in general
+    """
+    def is_device(self, account: str) -> bool:
+        """Ledger is always a device."""
+        return True
+
+    @validate_checksum_address
+    def unlock_account(self, account: str, password: str, duration: int = None) -> bool:
+        """Defer account unlocking to the ledger, do not indicate application level unlocking logic."""
+        return True
+
+    @validate_checksum_address
+    def lock_account(self, account: str) -> bool:
+        """Defer account locking to the ledger, do not indicate application level unlocking logic."""
+        return True
+
+
+# Fixup path regex in ledgereth to allow test derivation paths
+import ledgereth.utils
+ledgereth.utils.BIP32_ETH_PATTERN = r"^44'/(1|60)'/[0-9]+'(/[0-9]+){2}$"
+ledgereth.utils.BIP32_LEGACY_LEDGER_PATTERN = r"^44'/(1|60)'/[0-9]+'/[0-9]+$"
+
+
+class LedgerSigner(HardwareSigner):
+    """A ledger message and transaction signing client."""
+
+    __BIP44_TEST = "44'/1'/"
+    __BIP44_ETH = "44'/60'/"
+    __CHAIN_ID_MAINNET = 1
+    __CHAIN_ID_ROPSTEN = 3
+    __CHAIN_ID_RINKEBY = 4
+    __CHAIN_ID_GOERLI = 5
+    __CHAIN_ID_KOVAN = 42
+
+    def __init__(self, testnet: bool = False, count: int = 1, legacy: bool = False, path: str = None):
+        self.__addresses = dict()
+        self.__testnet = testnet
+
+        emitter = StdoutEmitter()
+        paths = (path,) if path else self.__get_paths(testnet, count, legacy)
+
+        for item in paths:
+            acct = ledger_get_account_by_path(item)
+            self.__addresses[acct.address] = acct
+            if testnet and acct.path.startswith(self.__BIP44_ETH):
+                raise self.SignerError(f'cannot use production path/account in testnet: {acct.path}/{acct.address}')
+            if not testnet and acct.path.startswith(self.__BIP44_TEST):
+                raise self.SignerError(f'cannot use testnet path/account in production: {acct.path}/{acct.address}')
+            emitter.message(f"Derived {acct.address} ({acct.path})")
```

Comment:
Do we need to print it here? Can it go to the log?

Reply:
> Do we need to print it here? Can it go to the log?

Personally, I like knowing when/what HW interactions are happening when using the CLI.  This is the same style of message we print for the trezor as well.

Reply:
FYI I modeled this after the trezor implementation hence the same key derivation messages.
Also some logic would likely be common to all walllet implementation (such as BIP32 key derivation logic, etc.) and could be factored out but I wanted this PR to be independent / isolated from other wallets to make code review easier, etc.

I'll be reworking this to implement EIP-155 transactions instead of the older style so I may need to ditch ledgereth (or include a modified version of it.)

Reply:
Thanks for following up on this issue - I has some kind of suspicion that this might be the case but have not yet had the opportunity to review it in-depth.